src/TuxState.js

```diff
@@ -0,0 +1,650 @@
+  //@param propsToAdd OBJECT: required object argument with deepest keys being numbers to subtract the current state from
+  //@param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  subtractState: function(propsToSubtract, callback) {
+    //throw error if propsToSubtract is undefined or not an object
+    invariantArgCheck(propsToSubtract);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    //build new state object
+    var newState = buildNewState(currentState, propsToSubtract, function(newState, key, currentStateAtKey, newPropsAtKey) {
+      //throw error if newPropsAtKey is not a number
+      invariantNumberCheck(newPropsAtKey);
+      //perform subtraction on the corresponding deep keys
+      newState[key] = currentStateAtKey - newPropsAtKey;
+    });
+    //set state with the updated values and a callback (if provided), triggering a re-render
+    this.setState(newState, callback);
+  },
+
+  //multiplyState FUNCTION: multiply the value of the deepest keys in the passed in object to the corresponding deepest keys in the current state, or throws an error if keys don't match
+  //@param propsToMultiply OBJECT: required object argument with deepest keys being numbers to multiply the current state by
+  //@param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  multiplyState: function (propsToMultiply, callback) {
+    //throw error if propsToMultiply is undefined or not an object
+    invariantArgCheck(propsToMultiply);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
+    //build new state object
+    var newState = buildNewState(currentState, propsToMultiply, function(newState, key, currentStateAtKey, newPropsAtKey) {
+      //throw error if newPropsAtKey is not a number
+      invariantNumberCheck(newPropsAtKey);
+      //perform multiplication on the corresponding deep keys
+      newState[key] = currentStateAtKey * newPropsAtKey;
+    });
+    //set state with the updated values and a callback (if provided), triggering a re-render
+    this.setState(newState, callback);
+  },
+
+  //divideState FUNCTION: divide the value of the deepest keys in the passed in object by the corresponding deepest keys in the current state, or throws an error if keys don't match
+  //@param propsToDivide OBJECT: required object argument with deepest keys being numbers to divide the current state by
+  //@param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  divideState: function (propsToDivide, callback) {
+    //throw error if propsToDivide is undefined or not an object
+    invariantArgCheck(propsToDivide);
+    //make a shallow copy of the current state
+    var currentState = assign({}, this.state);
```

Comment:
I don't think we need to invoke `assign` here because we are already making a shallow copy of state at the first line in `buildNewState`.